This patch is included to give an impression of the required changes and
in the hope that it will be easier to port to older or newer versions of
the kernel if that will be necessary in the future.

This patch was made against the FTDI SIO driver in linux-2.6.20.

--- ftdi_sio.h	2007-10-02 22:54:46.000000000 +0200
+++ ftdi_sio_dig528.h	2007-09-03 22:31:37.000000000 +0200
@@ -492,6 +492,12 @@
 #define FTDI_TACTRIX_OPENPORT_13S_PID	0xCC49	/* OpenPort 1.3 Subaru */
 #define FTDI_TACTRIX_OPENPORT_13U_PID	0xCC4A	/* OpenPort 1.3 Universal */
 
+/*
+ * Distlab dig528-2 boards.
+ */
+#define DISTLAB_VID		0x0050
+#define DISTLAB_DIG528_PID	0xC237
+
 /* Commands */
 #define FTDI_SIO_RESET 		0 /* Reset the port */
 #define FTDI_SIO_MODEM_CTRL 	1 /* Set the modem control register */
--- ftdi_sio.c	2007-10-02 22:54:46.000000000 +0200
+++ ftdi_sio_dig528.c	2007-10-02 23:00:15.000000000 +0200
@@ -17,6 +17,14 @@
  * See http://ftdi-usb-sio.sourceforge.net for upto date testing info
  *	and extra documentation
  *
+ * (03/Sep/2007) Jonas Fonseca
+ *      Cleaned up the driver hacks to require less changes so it will
+ *      be easier to port to newer kernel versions.
+ *
+ * (17/Nov/2006) Esben Zeuthen
+ *      Hacked the driver to use customized DIG-528 boards only.
+ *      Removed implicit DTR bit clear/set instructions.
+ *
  * (21/Jul/2004) Ian Abbott
  *      Incorporated Steven Turner's code to add support for the FT2232C chip.
  *      The prelimilary port to the 2.6 kernel was by Rus V. Brushkoff.  I have
@@ -258,14 +266,14 @@
 #include <linux/usb.h>
 #include <linux/serial.h>
 #include <linux/usb/serial.h>
-#include "ftdi_sio.h"
+#include "ftdi_sio_dig528.h"
 
 /*
  * Version Information
  */
 #define DRIVER_VERSION "v1.4.3"
-#define DRIVER_AUTHOR "Greg Kroah-Hartman <greg@kroah.com>, Bill Ryder <bryder@sgi.com>, Kuba Ober <kuba@mareimbrium.org>"
-#define DRIVER_DESC "USB FTDI Serial Converters Driver"
+#define DRIVER_AUTHOR "Greg Kroah-Hartman <greg@kroah.com>, Bill Ryder <bryder@sgi.com>, Kuba Ober <kuba@mareimbrium.org>, Esben Zeuthen <zept@diku.dk>"
+#define DRIVER_DESC "USB FTDI Serial Converters Driver hacked for DTR reset modified Freescale DIG528-2"
 
 static int debug;
 static __u16 vendor = FTDI_VID;
@@ -306,6 +314,8 @@
 
 
 static struct usb_device_id id_table_combined [] = {
+	{ USB_DEVICE(DISTLAB_VID, DISTLAB_DIG528_PID) },
+#if 0
 	{ USB_DEVICE(FTDI_VID, FTDI_AMC232_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_CANUSB_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_ACTZWAVE_PID) },
@@ -514,6 +524,7 @@
 	{ USB_DEVICE(FTDI_VID, FTDI_TACTRIX_OPENPORT_13S_PID) },
 	{ USB_DEVICE(FTDI_VID, FTDI_TACTRIX_OPENPORT_13U_PID) },
 	{ USB_DEVICE(ELEKTOR_VID, ELEKTOR_FT323R_PID) },
+#endif
 	{ },					/* Optional parameter entry */
 	{ }					/* Terminating entry */
 };
@@ -521,7 +532,7 @@
 MODULE_DEVICE_TABLE (usb, id_table_combined);
 
 static struct usb_driver ftdi_driver = {
-	.name =		"ftdi_sio",
+	.name =		"ftdi_sio_dig528",
 	.probe =	usb_serial_probe,
 	.disconnect =	usb_serial_disconnect,
 	.id_table =	id_table_combined,
@@ -612,7 +623,7 @@
 static struct usb_serial_driver ftdi_sio_device = {
 	.driver = {
 		.owner =	THIS_MODULE,
-		.name =		"ftdi_sio",
+		.name =		"ftdi_sio_dig528",
 	},
 	.description =		"FTDI USB Serial Device",
 	.id_table =		id_table_combined,
@@ -693,8 +704,8 @@
 	 return(ftdi_232bm_baud_base_to_divisor(baud, 48000000));
 }
 
-#define set_mctrl(port, set)		update_mctrl((port), (set), 0)
-#define clear_mctrl(port, clear)	update_mctrl((port), 0, (clear))
+#define set_mctrl(port, set)		update_mctrl((port), (set) & ~TIOCM_DTR, 0)
+#define clear_mctrl(port, clear)	update_mctrl((port), 0, (clear) & ~TIOCM_DTR)
 
 static int update_mctrl(struct usb_serial_port *port, unsigned int set, unsigned int clear)
 {
